### PR TITLE
[Bugfix] Email Preview Page Template Issue

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes fixes for template preview issues and cancels unnecessary queries with polished logic. Let me know your thoughts.

### CODE REVIEW

File: src/components/forms/MarkdownEditor.tsx [hunk 1]
```diff
@@ -96,6 +97,9 @@ export function MarkdownEditor(props: Props) {
           convert_urls: false,
         }}
         onEditorChange={handleChange}
+        onBlur={() => {
+          props.handleManualChange?.();
+        }}
```
[the remark this answers is not in the document]
Comment (Civolilah) [reply src/components/forms/MarkdownEditor.tsx:102]: @beganovich Yes, I agree - my mistake while changing solutions. It has been changed to handleOnBlur. Let me know your thoughts.

File: src/pages/invoices/email/components/Mailer.tsx [hunk 2]
```diff
@@ -51,62 +60,121 @@ interface Props {
...
-  const [ccEmail, setCcEmail] = useState('');
-  const reactSettings = useReactSettings();
+  const handleSend = useHandleSend({ setErrors });
+
+  const [payloadData, setPayloadData] = useState<Record<string, string>>({
+    body: '',
+    ccEmail: '',
+    subject: '',
+    templateId: props.defaultEmail,
+  });
```
[the remark this answers is not in the document]
Comment (Civolilah) [reply src/pages/invoices/email/components/Mailer.tsx:96]: @beganovich Yeh, sure thing, I've removed the type specification here.

**Comment by turbo124** [comment]
@Civolilah this looks good. And is certainly an upgrade that we can merge.

One edge case I do notice is that if we type very quickly and click send, I am able to beat the debounce. I am unsure if there is a more resilient way to handle these edge cases. by queueing the send behind a debounce also. ie, add a small debounce to the submit button that will allow the keystrokes to "catch up" ?

**Comment by Civolilah** [comment]
> @Civolilah this looks good. And is certainly an upgrade that we can merge.
> 
> One edge case I do notice is that if we type very quickly and click send, I am able to beat the debounce. I am unsure if there is a more resilient way to handle these edge cases. by queueing the send behind a debounce also. ie, add a small debounce to the submit button that will allow the keystrokes to "catch up" ?

@turbo124 I know what you're talking about. I guess you recreated it by using the tinymce field?

Actually in general, as you know, in the app we use the `onBlur` event in most cases to change values on fields, which is the case for cc_email and subject fields, but tinymce is a different case. There is a debounce of 500ms for changing values. From my perspective, to be honest, it's a little bit too much.

If you try to quickly change a value in this field and save it within 0.5 seconds in any part of the app, you will recreate the edge case issue (not saving last markdown editor value).

From my perspective, there is no need for delay when clicking the button, we just need to decrease this debounce when changing the value. On any plain text input field, debounce has been set to `300ms` when changing value without the onBlur event. So, I would suggest decreasing it to 300ms, same as for all other fields, and I believe it will be impossible to change value and move the mouse to click the button under 300ms to recreate the edge case issue. But everything will remain almost exactly the same.

It is always a fallback solution to add a small delay when clicking the save button, maybe a 100ms delay would avoid this edge case issue, but from my perspective, decreasing the debounce on the tinymce component is better.

Let me know your thoughts.

**Comment by turbo124** [comment]
@Civolilah thanks, perhaps we can test this in a new PR.

**Review by beganovich — APPROVED** [review]
